app/assets/javascripts/discourse/tests/acceptance/composer-test.js

```diff
@@ -956,6 +962,99 @@ acceptance("Composer", function (needs) {
       "it does not unescape script tags in code blocks"
     );
   });
+  function wait(timeout = 2000) {
+    return new Promise((resolve) => {
+      setTimeout(resolve, timeout);
+    });
+  }
+  test("Image alt text modification", async function (assert) {
+    // single image test case
+
+    const altText = ".image-wrapper .button-wrapper .alt-text";
+    const editAltTextButton =
+      ".image-wrapper .button-wrapper .alt-text-edit-btn";
+    const altTextInput = ".image-wrapper .button-wrapper .alt-text-input";
+
+    await visit("/");
+
+    await click("#create-topic");
+    await fillIn(".d-editor-input", `![zorro|200x200](upload://zorro.png)`);
+
+    // placement of elements
+
+    assert.ok(
+      exists(altText),
+      "shows alt text in the image wrapper's button wrapper"
+    );
+
+    assert.ok(
+      exists(editAltTextButton + " .d-icon-pencil"),
+      "alt text edit button with icon is in the image wrapper's button wrapper"
+    );
+
+    assert.ok(
+      exists(altTextInput),
+      "alt text input is in the image wrapper's button wrapper"
+    );
+
+    // logical
+
+    assert.equal(query(altText).innerText, "zorro", "correct alt text");
+    assert.ok(visible(altText), "alt text is visible");
+    assert.ok(visible(editAltTextButton), "alt text edit button is visible");
+    assert.ok(invisible(altTextInput), "alt text input is not visible");
+
+    await click(editAltTextButton);
+
+    assert.ok(invisible(altText), "readonly alt text is not visible");
+    assert.ok(
+      invisible(editAltTextButton),
+      "alt text edit button is not visible"
+    );
+    assert.ok(visible(altTextInput), "alt text input is visible");
+    assert.equal(
+      queryAll(altTextInput).val(),
+      "zorro",
+      "correct alt text in input"
+    );
+
+    await triggerKeyEvent(altTextInput, "keypress", "[".charCodeAt(0));
+    await triggerKeyEvent(altTextInput, "keypress", "]".charCodeAt(0));
+    assert.equal(
+      queryAll(altTextInput).val(),
+      "zorro",
+      "does not input [ ] keys"
+    );
+
+    await fillIn(altTextInput, "steak");
+    await triggerKeyEvent(altTextInput, "keypress", 13);
+
+    assert.equal(
+      queryAll(".d-editor-input").val(),
+      "![steak|200x200](upload://zorro.png)",
+      "alt text updated"
+    );
+    assert.equal(query(altText).innerText, "steak", "shows the alt text");
+    assert.ok(visible(editAltTextButton), "alt text edit button is visible");
+    assert.ok(invisible(altTextInput), "alt text input is not visible");
+
+    // multiple image test case
```

Comment:
Actually caught a sneaky bug with the multiple image test case. Putting this test case below the 'single image' one since the query selectors are slightly annoying.

Reply:
You could create some setup function, and use it in two tests